src/main/java/org/asamk/signal/ReceiveMessageHandler.java

```diff
@@ -421,16 +421,119 @@ private void handleSignalServiceDataMessage(SignalServiceDataMessage message) {
+                }
+                if (name.getFamily().isPresent() && !name.getFamily().get().isBlank()) {
+                    System.out.println("   - Family name: " + name.getFamily().get());
+                }
+                if (name.getPrefix().isPresent() && !name.getPrefix().get().isBlank()) {
+                    System.out.println("   - Prefix name: " + name.getPrefix().get());
+                }
+                if (name.getSuffix().isPresent() && !name.getSuffix().get().isBlank()) {
+                    System.out.println("   - Suffix name: " + name.getSuffix().get());
+                }
+
+                if (contact.getAvatar().isPresent()) {
+                    SharedContact.Avatar avatar = contact.getAvatar().get();
+                    System.out.println(" - Avatar:");
+                    printAttachment(avatar.getAttachment(), 3);
+                    if (avatar.isProfile()) {
+                        System.out.println("   - Is profile");
+                    } else {
+                        System.out.println("   - Is not a profile");
+                    }
+                }
+
+                if (contact.getPhone().isPresent()) {
+                    System.out.println(" - Phone details:");
+                    int i = 0;
+                    for (SharedContact.Phone phone : contact.getPhone().get()) {
+                        System.out.println("   - No. " + i + ":");
+                        if (phone.getValue() != null) {
+                            System.out.println("     - Number: " + phone.getValue());
+                        }
+                        if (phone.getType() != null) {
+                            System.out.println("     - Type: " + phone.getType());
+                        }
+                        if (phone.getLabel().isPresent() && !phone.getLabel().get().isBlank()) {
+                            System.out.println("     - Label: " + phone.getLabel().get());
+                        }
+                        i++;
+                    }
+                }
+
+                if (contact.getEmail().isPresent()) {
+                    System.out.println(" - Email details:");
+                    int i = 0;
+                    for (SharedContact.Email email : contact.getEmail().get()) {
+                        System.out.println("   - No. " + i + ":");
+                        if (email.getValue() != null) {
+                            System.out.println("     - Value: " + email.getValue());
+                        }
+                        if (email.getType() != null) {
+                            System.out.println("     - Type: " + email.getType());
+                        }
+                        if (email.getLabel().isPresent() && !email.getLabel().get().isBlank()) {
+                            System.out.println("     - Label: " + email.getLabel().get());
+                        }
+                    }
```

Comment:
missing `i++`
though I'm not sure if numbering the lists is very useful anyway ...

Reply:
Thanks. The idea behind it is the `"  - No. " + i` line is used to separate the different Phones/Emails/Addresses otherwise you get output like this
```
 - Email details:
    - Value: x1
    - Type: y1
    - Label: z1
    - Value: x2
    - Type: y2
    - Label: z2
```
Maybe I can change the `"  - No. " + i` to instead be "  - Email: " or some other message.